### PR TITLE
pipelineymlgen: add ymlrange, clarify inlinerange

### PR DESCRIPTION
**Pull request by dagood**
This pull request introduces a new `ymlrange` command. The `ymlrange` command is similar to `inlinerange` but doesn't (generally speaking) merge iteration results into the parent element.

This allows for this use, in a more intentional way than [microsoft/go-infra#458](https://github.com/microsoft/go-infra/pull/458), so it doesn't cut off other potential use of these expressions in the future:

This allows for this use:

```yml
    containers:
      ${ ymlrange .containers }:
        ${ .container }:
          image: ${ .image }
```

```yml
    containers:
      ubuntu2204:
        image: mcr.microsoft.com/microsoft-go/infra-images:ubuntu-22.04-amd64-default
      mariner2:
        image: mcr.microsoft.com/microsoft-go/infra-images:cbl-mariner-2.0-amd64-default
```

Adding a new expression lets them both be used in a more intentional way than [microsoft/go-infra#458](https://github.com/microsoft/go-infra/pull/458). This way, we don't cut off some potential other uses.

I think there may still be some unclear edge cases. Might need to tighten it down by adding more test cases in the future. For now, probably good enough.

All but `ymlrange-multi-key` test case authored by Copilot.

### CODE REVIEW

File: internal/pipelineymlgen/eval.go [hunk 1]
```diff
@@ -449,6 +450,16 @@ func (e *EvalState) mappingToYAML(m *evalMapping) (*yaml.Node, error) {
...
 			}
 
+			// ymlrange produces a single merged node that should not
+			// be flattened into the surrounding mapping. Return it
+			// directly so the parent can treat it as one element.
+			if er, ok := a.(*evalRange); ok && !er.inline {
+				if n.Kind != 0 {
+					return fail(fmt.Errorf("ymlrange cannot be combined with other mapping content"))
+				}
+				return node, nil
```
[the remark this answers is not in the document]
Comment (dagood) [reply internal/pipelineymlgen/eval.go]: @copilot add a test case and fix this.

[the remark this answers is not in the document]
Comment (dagood) [reply internal/pipelineymlgen/eval.go]: Hmm. No, it doesn't make sense that keys can't be inserted into a map.
